alacritty/src/display/mod.rs

```diff
@@ -366,179 +374,89 @@ pub struct Display {
-        let font = &config.font;
-        let rasterizer = Rasterizer::new(estimated_scale_factor as f32)?;
-        let mut glyph_cache = GlyphCache::new(rasterizer, font)?;
+        let mut glyph_cache = GlyphCache::new(rasterizer, &config.font)?;
+
         let metrics = glyph_cache.font_metrics();
         let (cell_width, cell_height) = compute_cell_size(config, &metrics);
 
-        // Guess the target window size if the user has specified the number of lines/columns.
-        let dimensions = config.window.dimensions();
-        let estimated_size = dimensions.map(|dimensions| {
-            window_size(config, dimensions, cell_width, cell_height, estimated_scale_factor)
-        });
+        // Resize the window to account for user size.
+        if let Some(dimensions) = config.window.dimensions() {
+            let size = window_size(config, dimensions, cell_width, cell_height, scale_factor);
+            window.set_inner_size(size);
+        }
 
-        debug!("Estimated scaling factor: {}", estimated_scale_factor);
-        debug!("Estimated window size: {:?}", estimated_size);
-        debug!("Estimated cell size: {} x {}", cell_width, cell_height);
+        let window_inner_size = window.inner_size();
 
-        // Spawn the Alacritty window.
-        let window = Window::new(
-            event_loop,
-            config,
-            identity,
-            estimated_size,
-            #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
-            wayland_event_queue,
-        )?;
+        // Get the display and the config used to create that context.
+        let gl_display = context.display();
+        let gl_config = context.config();
 
-        // Create renderer.
-        let mut renderer = Renderer::new()?;
+        let surface_attributes =
+            SurfaceAttributesBuilder::<WindowSurface>::new().with_srgb(Some(false)).build(
+                window.raw_window_handle(),
+                NonZeroU32::new(window_inner_size.width).unwrap(),
+                NonZeroU32::new(window_inner_size.height).unwrap(),
+            );
 
-        let scale_factor = window.scale_factor;
-        info!("Display scale factor: {}", scale_factor);
+        // Create the GL surface to draw into.
+        let surface = unsafe { gl_display.create_window_surface(&gl_config, &surface_attributes)? };
 
-        // If the scaling factor changed update the glyph cache and mark for resize.
-        let should_resize = (estimated_scale_factor - window.scale_factor).abs() > f64::EPSILON;
-        let (cell_width, cell_height) = if should_resize {
-            Self::update_font_size(&mut glyph_cache, scale_factor, config, font)
-        } else {
-            (cell_width, cell_height)
-        };
+        // Make the context current.
+        let context = context.make_current(&surface)?;
+
+        // We need to load them only once per instance, but only after we make our context current
+        // due to WGL limitations.
```

Comment:
```suggestion
        // We need to load OpenGL functions once per instance, but only after we make our context current
        // due to WGL limitations.
```

"them" is kinda confusing if you read the comment before looking at the code.